### PR TITLE
typescript tests and fixes

### LINKED ISSUE #7: Build 1.0.3 is missing typescript index.d.ts
### Are you submitting a **bug report** or a **feature request**?
bug

### What is the current behavior?
No ts bindings are in the build

### What is the expected behavior?
source `index.d.ts` is built and packaged as `"typings": "dist/index.d.js"`

### Sandbox Link
https://unpkg.com/final-form-arrays@1.0.3/dist/index.d.js

### PR DESCRIPTION
**Pull request by rosskevin**
I improved the structure of the typescript exports (now strongly typed) and added tests for all the functions.

I _believe_ I found the problem with #7 in that the package was referring to a `js` file instead of a `ts` file.

@erikras Please try a build and see if the `ts` gets included.

Closes #7 
Depends on https://github.com/final-form/final-form/pull/68

### CODE REVIEW

File: src/index.d.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const onSubmit: Config['onSubmit'] = (values, callback) => {}
+
+const form = createForm({
+  mutators: { ...arrayMutators },
+  onSubmit
+})
+
+// Get form.mutators (default as object) and cast to Mutators
+const mutators: Mutators = ((form.mutators || {}) as any) as Mutators
```
Comment (rosskevin) [inline src/index.d.test.ts]: FYI - This casting is a bit gross, but unless we allow parameterized types in the form and expose it all the way through, this is the way it needs to be done.  Parameterized types in jsx is already painful so this is not any worse by any means.

**Comment by rosskevin** [comment]
I need to PR a change to `react-final-form` to `FormRenderProps` `mutators` to make this simpler (make sure they are not undefined).   The test I have created in https://github.com/final-form/react-final-form-arrays/pull/18 show that this will be really painful in a strongly typed application.

**Comment by erikras** [comment]
Okay. I upgraded the final-form dep. Let me know when this PR is good to go.

**Comment by rosskevin** [comment]
@erikras This one is good to merge/release

**Comment by erikras** [comment]
Published in [`v1.0.4`](https://github.com/final-form/final-form-arrays/releases/tag/v1.0.4).

**Pull request merged**